### PR TITLE
executor:add more comments for baseExecutor

### PR DESCRIPTION
**Pull request by zhexuany**
This PR aims to add more comments in baseExecutor.

### CODE REVIEW

**Comment by zhexuany** [comment]
@zz-jason @coocood PTAL

File: executor/executor.go [hunk 1]
```diff
@@ -107,7 +108,7 @@ type baseExecutor struct {
 	retFieldTypes   []*types.FieldType
 }
 
-// Open implements the Executor Open interface.
+// Open opens all executors and initializes childrenResults according to executors' length.
```
Comment (zz-jason) [inline executor/executor.go]: // Open initializes children recursively and "childrenResults" according to children's schemas.

File: executor/executor.go [hunk 2]
```diff
@@ -97,6 +97,7 @@ const (
 // Otherwise the executor's returned rows don't need to store the handle information.
 type Row = types.DatumRow
 
+// baseExecutor contains an Executor's array and it implements visitor pattern.
```
Comment (zz-jason) [inline executor/executor.go]: // baseExecutor contains the common resource needed by every "Executor", including children "Executor"s, execution context, schema.

Comment (zz-jason) [reply executor/executor.go]: It has not implemented the visitor pattern.

File: executor/executor.go [hunk 3]
```diff
@@ -122,7 +123,7 @@ func (e *baseExecutor) Open(goCtx goctx.Context) error {
 	return nil
 }
 
-// Close implements the Executor Close interface.
+// Close closes all executors and assigns nil to childrenResults for clearing all reference pointing to it.
```
Comment (zz-jason) [inline executor/executor.go]: s/assigns nil/release all resource/

File: executor/executor.go [hunk 4]
```diff
@@ -163,6 +165,7 @@ func (e *baseExecutor) supportChunk() bool {
 	return true
 }
 
+// NextChunk fills a chunk, which is passed by referenbce, with mutiple rows.
```
Comment (shenli) [inline executor/executor.go]: referenbce？

Comment (coocood) [reply executor/executor.go]: s/mutiple/multiple

File: executor/executor.go (hunk 2 above)
Comment (zhexuany) [reply executor/executor.go]: Per my understanding, that is visitor pattern.

File: executor/executor.go [hunk 5]
```diff
@@ -122,7 +123,7 @@ func (e *baseExecutor) Open(goCtx goctx.Context) error {
 	return nil
 }
 
-// Close implements the Executor Close interface.
+// Close closes all executors and release all resource.
```
Comment (coocood) [inline executor/executor.go]: release all the resources.

File: executor/executor.go [hunk 6]
```diff
@@ -134,19 +135,20 @@ func (e *baseExecutor) Close() error {
 	return nil
 }
 
-// Schema implements the Executor Schema interface.
+// Schema returns current baseExecutor's schema. If its value is nil, then create a new one.
```
Comment (coocood) [inline executor/executor.go]: the current

Comment (zhexuany) [reply executor/executor.go]: actually, I feel like  either "the baseExecutor's schema" or "current baseExecutor's schema". "the current" make me thinking it's duplicate meaning such as "In addition, we also implement...."

File: executor/executor.go [hunk 7]
```diff
@@ -163,6 +165,7 @@ func (e *baseExecutor) supportChunk() bool {
 	return true
 }
 
+// NextChunk fills a chunk, which is passed by reference, with multiple rows.
```
Comment (coocood) [inline executor/executor.go]: Add the comment to the `Executor` interface is better.

File: executor/executor.go (hunk 5 above)
Comment (zhexuany) [reply executor/executor.go]: Well, I was thinking "resource" is a uncountable noun.

File: executor/executor.go (hunk 7 above)
Comment (zhexuany) [reply executor/executor.go]: DONE

**Comment by coocood** [comment]
LGTM

**Comment by zhexuany** [comment]
@zz-jason PTAL again. I just addressed your comments.

File: executor/executor.go [hunk 8]
```diff
@@ -134,19 +135,20 @@ func (e *baseExecutor) Close() error {
 	return nil
 }
 
-// Schema implements the Executor Schema interface.
+// Schema returns the current baseExecutor's schema. If its value is nil, then create a new one.
```
Comment (zz-jason) [inline executor/executor.go]: If it is nil, an empty new "Schema" is returned.

File: executor/executor.go [hunk 9]
```diff
@@ -134,19 +135,20 @@ func (e *baseExecutor) Close() error {
...
 func (e *baseExecutor) Schema() *expression.Schema {
 	if e.schema == nil {
 		return expression.NewSchema()
 	}
 	return e.schema
 }
 
+// newChunk creates a new chunk for future use.
```
Comment (zz-jason) [inline executor/executor.go]: creates a "Chunk" to buffer the result of current executor.

File: executor/executor.go (hunk 2 above)
Comment (zz-jason) [reply executor/executor.go]: But it's different from the traditional, common used visitor pattern, witch may further cause some misunderstanding.

**Comment by zz-jason** [comment]
@zhexuany I think it's more worthy to write comments for `Executor` interface. take this as an example:
https://github.com/cloudera/Impala/blob/cdh5-trunk/be/src/exec/exec-node.h

**Pull request merged**